src/core/evaluator.js

```diff
@@ -1011,14 +1011,18 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
 
           var tx = 0;
           var ty = 0;
+          var charSpacing = 0;
+          if (i > 0 || textChunk.str.length > 0) {
```

Comment:
Nit: perhaps add a comment here explaining the logic, e.g. something along the lines of https://github.com/mozilla/pdf.js/issues/5972#issuecomment-97458306.